src/xo-app/settings/plugins/index.js

```diff
@@ -120,7 +121,13 @@ class Plugin extends Component {
     this.refs.pluginInput.value = this.props.configurationPresets[configName]
   }
 
-  _test = async () => testPlugin(this.props.id, this.refs.testInput.value)
+  _test = async () => {
+    try {
+      await testPlugin(this.props.id, this.refs.testInput.value)
+    } catch (err) {
+      alert('You have an error!', err.stack)
```

Comment:
You should wait for modal to be closed, `alert()` is an async method which resolves when the modal closes.